Flaky workbox-google-analytics test

**Library Affected**:
*workbox-google-analytics*

I'm seeing periodic flakiness when running the `workbox-google-analytics` integration tests, against either the `prod` or `dev` bundle, in the current Google Chrome Stable (M70).

```sh
1) [workbox-google-analytics] Load and use Google Analytics
       replay failed Google Analytics hits:

      AssertionError: expected [ Array(1) ] to have a length of 2 but got 1
      + expected - actual

      -1
      +2

      at Context.<anonymous> (test/workbox-google-analytics/integration/basic-example.js:125:30)
      at process._tickCallback (internal/process/next_tick.js:68:7)
```

--- a/package.json
+++ b/package.json
@@ -102,7 +102,7 @@
     "semver": "^5.5.1",
     "serve-index": "^1.9.1",
     "service-worker-mock": "^1.9.3",
-    "shelving-mock-indexeddb": "github:philipwalton/shelving-mock-indexeddb#621b7a275568051846d20e3e557fa1101418b1d1",
+    "shelving-mock-indexeddb": "github:philipwalton/shelving-mock-indexeddb#c7b3b002472597ee75b027011049737a06460261",
     "sinon": "^6.3.4",
     "tempy": "^0.2.1",
     "url-search-params": "^1.1.0",

--- a/packages/workbox-background-sync/models/QueueStore.mjs
+++ b/packages/workbox-background-sync/models/QueueStore.mjs
@@ -91,9 +91,12 @@ export class QueueStore {
       });
     }
 
-    const firstEntry = await this._db.get(OBJECT_STORE_NAME);
+    const [firstEntry] = await this._db.getAllMatching(OBJECT_STORE_NAME, {
+      count: 1,
+    });
+
     if (firstEntry) {
-    // Pick an ID one less than the lowest ID in the object store.
+      // Pick an ID one less than the lowest ID in the object store.
       entry.id = firstEntry.id - 1;
     } else {
       delete entry.id;

--- a/packages/workbox-google-analytics/initialize.mjs
+++ b/packages/workbox-google-analytics/initialize.mjs
@@ -47,7 +47,8 @@ const createOnSyncCallback = (config) => {
         // Measurement protocol requests can set their payload parameters in
         // either the URL query string (for GET requests) or the POST body.
         const params = request.method === 'POST' ?
-            new URLSearchParams(await request.text()) : url.searchParams;
+            new URLSearchParams(await request.clone().text()) :
+            url.searchParams;
 
         // Calculate the qt param, accounting for the fact that an existing
         // qt param may be present and should be updated rather than replaced.
@@ -70,7 +71,7 @@ const createOnSyncCallback = (config) => {
           config.hitFilter.call(null, params);
         }
 
-        // Retry the fetch. Ignore URL search params form the URL as they're
+        // Retry the fetch. Ignore URL search params from the URL as they're
         // now in the post body.
         await fetch(new Request(url.origin + url.pathname, {
           body: params.toString(),

--- a/test/workbox-core/node/_private/test-DBWrapper.mjs
+++ b/test/workbox-core/node/_private/test-DBWrapper.mjs
@@ -193,15 +193,18 @@ describe(`DBWrapper`, function() {
             const txn = evt.target.transaction;
 
             const objStore = txn.objectStore('users');
-            objStore.get().onsuccess = (event) => {
-              const firstUser = event.target.result;
-              // Delete the object store and recreate it with new config.
-              db.deleteObjectStore('users');
-              db.createObjectStore('users', {
-                keyPath: 'id',
-                autoIncrement: true,
-              });
-              objStore.add(firstUser);
+            objStore.openCursor().onsuccess = ({target}) => {
+              const cursor = target.result;
+              if (cursor) {
+                const firstUser = cursor.value;
+                // Delete the object store and recreate it with new config.
+                db.deleteObjectStore('users');
+                db.createObjectStore('users', {
+                  keyPath: 'id',
+                  autoIncrement: true,
+                });
+                objStore.add(firstUser);
+              }
             };
           }
         },

